reporters/reporter-default/src/main/java/org/radargun/reporting/html/HtmlReporter.java

```diff
@@ -147,47 +172,109 @@ private void writeTestReportDocuments(Set<String> combinedTests, Map<String, Lis
+
+            Map root = new HashMap();
+            root.put("timelineDocument", timelineDocument);
+            reportTemplate.process(root, printWriter);
+            printWriter.close();
+         } catch (Exception e) {
+            log.error("Templating exception", e);
          }
       }
    }
 
    private void writeIndexDocument(Collection<Report> reports, Set<String> allTests) {
       IndexDocument index = new IndexDocument(targetDir);
+      index.createReportDirectory();
+      index.prepareDocument(reports);
+      freemarker.template.Configuration cfg = initConfig();
       try {
-         index.open();
-         index.writeTests(allTests);
-         index.writeConfigurations(reports);
-         index.writeScenario(reports);
-         index.writeTimelines(reports);
-         index.writeFooter();
-      } catch (IOException e) {
-         log.error("Failed to write HTML report.", e);
-      } finally {
-         index.close();
+         Template indexTemplate = cfg.getTemplate("index.ftl");
+         PrintWriter printWriter = new PrintWriter(targetDir + File.separator + "index.html");
+
+         Map root = new HashMap();
+         root.put("reporter", this);
+         root.put("indexDocument", index);
+
+         copyResources("style.css");
+         copyResources("script.js");
+         indexTemplate.process(root, printWriter);
+         printWriter.close();
+      } catch (Exception e) {
+         log.error("Templating exception", e);
       }
    }
+
+   private void copyResources(String file) throws IOException {
+      ClassLoader classLoader = getClass().getClassLoader();
+      InputStream source = classLoader.getResourceAsStream("html/templates/" + file);
+      File destination = new File(targetDir + File.separator + file);
+      Files.copy(source, destination.toPath(), StandardCopyOption.REPLACE_EXISTING);
+
+   }
+
+   public static freemarker.template.Configuration initConfig() {
+      freemarker.template.Configuration cfg = new freemarker.template.Configuration(freemarker.template.Configuration.VERSION_2_3_23);
+      cfg.setTemplateLoader(new ClassTemplateLoader(HtmlReporter.class, "/html/templates"));
+
+      DefaultObjectWrapperBuilder builder = new DefaultObjectWrapperBuilder(freemarker.template.Configuration.VERSION_2_3_23);
+      builder.setExposeFields(true);
+      DefaultObjectWrapper bw = builder.build();
+      cfg.setAPIBuiltinEnabled(true);
+      // setting how booleans are displayed, default results in error
+      cfg.setBooleanFormat("True, False");
+      cfg.setObjectWrapper(bw);
+      return cfg;
+   }
+   // These methods are used in templates
+
+   public String getSystemProperty(String property) {
+      return System.getProperty(property);
+   }
+
+   public Set<String> getAllTests() {
+      return allTests;
+   }
+
+   public Collection<Report> getReports() {
+      return reports;
+   }
+
+   public String getTargetDir() {
```

Comment:
I'm removing getTargetDir() method, the rest is used in template index.ftl
e.g. getAllTests() method can be referred to from template as reporter.getAllTests() or reporter.allTests